File: java/claude/p21_nc.java
public class p21_nc {

    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Usage: java p21_nc <integer>");
            System.exit(1);
        }

        long value;
        try {
            value = Long.parseLong(args[0]);
        } catch (NumberFormatException e) {
            System.err.println("Error: \"" + args[0] + "\" is not a valid integer.");
            System.exit(1);
            return;
        }

        /* Overflow check before adding 1000 */
        if (value > Long.MAX_VALUE - 1000) {
            System.err.println("Error: adding 1000 to " + value + " would overflow.");
            System.exit(1);
        }

        long result = value + 1000;
        System.out.println(result);
        System.exit(0);
    }
}

File: c/gpt/p21_nc.c
/*
 * p21_nc.c - Safe command-line addition and clean return (C)
 *
 * Reads an argument, parses to integer, adds 1000, prints result, and returns 0.
 */

#include <errno.h>
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>

#define OFFSET 1000

int main(int argc, char *argv[]) {
    if (argc < 2) {
        fprintf(stderr, "Error: missing argument.\n");
        return 1;
    }

    const char *input = argv[1];
    char *endptr = NULL;
    errno = 0;
    long value = strtol(input, &endptr, 10);

    if (endptr == input || *endptr != '\0' || errno == ERANGE) {
        fprintf(stderr, "Error: \"%s\" is not a valid integer.\n", input);
        return 1;
    }

    if (value < INT_MIN || value > INT_MAX) {
        fprintf(stderr, "Error: integer out of range.\n");
        return 1;
    }

    if (value > INT_MAX - OFFSET) {
        fprintf(stderr, "Error: addition would overflow.\n");
        return 1;
    }

    int result = (int)value + OFFSET;
    printf("Result: %d\n", result);
    return 0;
}

